lib/min_heap.rb

```diff
@@ -55,17 +61,46 @@ def empty?
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: o(1)
+  # Space complexity: o(1)
   def heap_up(index)
-    
+    return if index == 0
+    parent_index = (index % 2 != 0) ? (index / 2) : (index / 2 - 1)
+    if @store[index].key < @store[parent_index].key
+      swap(index,parent_index)
+      heap_up(parent_index)
+    else
+      return
+    end
   end
 
   # This helper method takes an index and 
-  #  moves it up the heap if it's smaller
-  #  than it's parent node.
+  #  moves it down the heap if it's bigger
+  #  than its child node.
   def heap_down(index)
```

Comment:
This works, but may be making extra recursive calls.  You should only continue to heap down if you made a swap.